### ISSUE TITLE
Can't dinamically change both data and colors in barchart.

### ISSUE THREAD

**Issue by caph1993** [valor-software/ng2-charts#1086]
I have a barplot.
* If I update the colors with a custom button, it works.
* If I update the data with a custom button, it works.
* If I update the data and the colors, the data changes, but the colors don't change.

```html
  <canvas baseChart
  [chartType]="bar"
  [datasets]="data"
  [colors]="chartColors"
  </canvas>
```

```typescript
  private data = [];
  update_data(){
    this.data = this.Y.map((y, i)=>({
      data: y,
      label: this.labels[i],
    }));
    this.chartColors[0].backgroundColor = this.colors.slice();
    this.chartColors = this.chartColors.slice();
  }
```
Just pick some random this.colors and this._Y, and you'll see the bug.

**Comment by caph1993** [comment]
There's a workaround that works: Changing the values inside `this.data` without assigning a new object to it:

```typescript
  assign(x,y){
    x.splice(0, x.length);
    x.push(...y);
  }

  private data = [];
  update_data(){
    this._Y.forEach((y, i)=>{
      if(i==this.data.length) this.data.push({data:[], label:''});
      this.assign(this.data[i].data, y);
      this.data[i].label=this._labels[i];
    });
    this.chartColors[0].backgroundColor = this.colors.slice();
    this.chartColors = this.chartColors.slice();
  }
```

**Comment by paviad** [comment]
This should be resolved in version 2.1.0.

Closed by ec2985d
